app/javascript/controllers/tab_controller.js

```diff
@@ -3,14 +3,21 @@ import { Controller } from "@hotwired/stimulus";
 // Connects to data-controller="tab"
 export default class extends Controller {
   static classes = ["active", "inactive"];
-  static targets = ["day"];
+  static targets = ["dayPrediction", "day", "daqiValue"];
 
   connect() {}
 
   switch_tab() {
     this.makeAllTabsInactive();
-    this.dayTarget.classList.toggle(this.activeClass);
-    this.dayTarget.classList.toggle(this.inactiveClass);
+    this.removeDaqiAlertClasses();
+    if (
+      this.dayTarget.innerText !== "Today" &&
+      this.daqiValueTarget.innerText[6] > 3
+    ) {
+      this.addDaqiAlertClass();
+    }
```

Comment:
this might perhaps be clearer if you extracted to a function with an intention-revealing name e.g. `addAlertClassIfTodayHasAirQualityAlert()`?

Reply:
yes, although it's more like `addAlertClassIfNotTodayAndHasAirQualityAlert` 😄

Reply:
maybe that is ok though, even if it is long